src/coredump/coredump.c

```diff
@@ -919,7 +918,11 @@ static int process_journald_crash(const char *context[], int input_fd) {
         if (r < 0)
                 return r;
 
-        log_info("Detected coredump of the journal daemon itself, diverted to %s.", filename);
+        log_notice("Detected coredump of the journal daemon or PID 1, diverted to %s.", filename);
+
+        log_notice("Due to the special circumstances, coredump collection will now be turned off.");
+        (void) write_string_file("/proc/sys/kernel/core_pattern", "|/bin/false", 0);
```

Comment:
Should we overwrite the `/proc/sys/kernel/core_pattern` on the `systemd-journald` crash?
```sh
-bash-4.3# cat /proc/sys/kernel/core_pattern
|/usr/lib/systemd/systemd-coredump %P %u %g %s %t %c %e

-bash-4.3# systemctl show --property=MainPID --value --no-pager systemd-journald
247

-bash-4.3# ps 247
  PID TTY      STAT   TIME COMMAND
  247 ?        Ss     0:01 /usr/lib/systemd/systemd-journald

-bash-4.3# kill -ABRT 247

-bash-4.3# cat /proc/sys/kernel/core_pattern
|/bin/false
```
I think we shouldn't